config/env_spec.go

```diff
@@ -602,3 +611,101 @@ type CorsPolicy struct {
 func (c CorsPolicy) IsEmpty() bool {
 	return len(c.AllowOrigins) == 0 && len(c.AllowOriginsRegexes) == 0
 }
+
+// TargetAuthentication configures how to authenticate the request to the backend.
+type TargetAuthentication struct {
+	// The time between two adjacent token refreshments.
+	RefreshInterval time.Duration `yaml:"refresh_interval,omitempty" mapstructure:"refresh_interval,omitempty"`
+
+	// GoogleOAuth configures how to authenticate the request to the backend with Google OAuth tokens.
+	GoogleOAuth GoogleOAuth `yaml:"google_oauth,omitempty" mapstructure:"google_oauth,omitempty"`
```

Comment:
Should this struct be a oneof, so it could be a GoogleOAuth or another kind? Maybe something like OAuthProvider or something like that?